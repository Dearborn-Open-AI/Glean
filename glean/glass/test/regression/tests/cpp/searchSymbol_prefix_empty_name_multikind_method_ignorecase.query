action: searchSymbol
args: {"name": "", "repo_name": "test", "language": [1], "kinds": [8], "options": {"detailedResults": false, "exactMatch": false, "ignoreCase": true, "namespaceSearch": false } }
