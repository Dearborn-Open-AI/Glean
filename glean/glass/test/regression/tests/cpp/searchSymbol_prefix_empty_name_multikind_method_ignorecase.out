[
    "@generated",
    {
        "symbolDetails": [],
        "symbols": [
            {
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 1,
                        "columnEnd": 2,
                        "lineBegin": 205,
                        "lineEnd": 207
                    },
                    "repository": "test"
                },
                "name": "bar",
                "qname": {
                    "container": "N1::N2::Bar",
                    "localName": "bar(N1::N2::N3::Foo)"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//N1/N2/Bar/bar/.f/N1::N2::N3::Foo"
            },
            {
                "context": {
                    "qname": {
                        "container": "N1::N2",
                        "localName": "Bar"
                    },
                    "symbol": "test/cpp//N1/N2/Bar"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 27,
                        "lineBegin": 188,
                        "lineEnd": 188
                    },
                    "repository": "test"
                },
                "name": "bar",
                "qname": {
                    "container": "N1::N2::Bar",
                    "localName": "bar(N1::N2::N3::Foo)"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//N1/N2/Bar/bar/.f/N1::N2::N3::Foo/.decl"
            },
            {
                "context": {
                    "qname": {
                        "container": "Q",
                        "localName": "QQ"
                    },
                    "symbol": "test/cpp//Q/QQ"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 17,
                        "lineBegin": 135,
                        "lineEnd": 135
                    },
                    "repository": "test"
                },
                "name": "bozzo",
                "qname": {
                    "container": "Q::QQ",
                    "localName": "bozzo"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//Q/QQ/bozzo/.f//.decl"
            },
            {
                "context": {
                    "qname": {
                        "container": "Q",
                        "localName": "QQ"
                    },
                    "symbol": "test/cpp//Q/QQ"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 30,
                        "lineBegin": 137,
                        "lineEnd": 137
                    },
                    "repository": "test"
                },
                "name": "getDouble",
                "qname": {
                    "container": "Q::QQ",
                    "localName": "getDouble() const&"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//Q/QQ/getDouble/.f//const,lvalue/.decl"
            },
            {
                "context": {
                    "qname": {
                        "container": "Q",
                        "localName": "QQ"
                    },
                    "symbol": "test/cpp//Q/QQ"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 26,
                        "lineBegin": 138,
                        "lineEnd": 138
                    },
                    "repository": "test"
                },
                "name": "getDouble",
                "qname": {
                    "container": "Q::QQ",
                    "localName": "getDouble() &"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//Q/QQ/getDouble/.f//lvalue/.decl"
            },
            {
                "context": {
                    "qname": {
                        "container": "Q",
                        "localName": "QQ"
                    },
                    "symbol": "test/cpp//Q/QQ"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 26,
                        "lineBegin": 139,
                        "lineEnd": 139
                    },
                    "repository": "test"
                },
                "name": "getDouble",
                "qname": {
                    "container": "Q::QQ",
                    "localName": "getDouble() &&"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//Q/QQ/getDouble/.f//rvalue/.decl"
            },
            {
                "context": {
                    "qname": {
                        "container": "Q",
                        "localName": "QQ"
                    },
                    "symbol": "test/cpp//Q/QQ"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 29,
                        "lineBegin": 143,
                        "lineEnd": 143
                    },
                    "repository": "test"
                },
                "name": "thango",
                "qname": {
                    "container": "Q::QQ",
                    "localName": "thango(int, Q::QQ &&) &&"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//Q/QQ/thango/.f/int,Q::QQ+%26%26/rvalue/.decl"
            },
            {
                "context": {
                    "qname": {
                        "container": "Q",
                        "localName": "QQ"
                    },
                    "symbol": "test/cpp//Q/QQ"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 28,
                        "lineBegin": 142,
                        "lineEnd": 142
                    },
                    "repository": "test"
                },
                "name": "thango",
                "qname": {
                    "container": "Q::QQ",
                    "localName": "thango(int, Q::QQ &) &"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//Q/QQ/thango/.f/int,Q::QQ+%26/lvalue/.decl"
            },
            {
                "context": {
                    "qname": {
                        "container": "Q",
                        "localName": "QQ"
                    },
                    "symbol": "test/cpp//Q/QQ"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 38,
                        "lineBegin": 141,
                        "lineEnd": 141
                    },
                    "repository": "test"
                },
                "name": "thango",
                "qname": {
                    "container": "Q::QQ",
                    "localName": "thango(int, const Q::QQ &) const&"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//Q/QQ/thango/.f/int,const+Q::QQ+%26/const,lvalue/.decl"
            },
            {
                "context": {
                    "qname": {
                        "container": "Q",
                        "localName": "QQ"
                    },
                    "symbol": "test/cpp//Q/QQ"
                },
                "kind": 8,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 5,
                        "columnEnd": 43,
                        "lineBegin": 145,
                        "lineEnd": 145
                    },
                    "repository": "test"
                },
                "name": "thingo",
                "qname": {
                    "container": "Q::QQ",
                    "localName": "thingo(pear<int, char>, Q::QQ &&) const"
                },
                "score": {
                    "exactness": 2
                },
                "symbol": "test/cpp//Q/QQ/thingo/.f/pear%3Cint%20+char%3E,Q::QQ+%26%26/const/.decl"
            }
        ]
    }
]